### PR TITLE
Tpp+ven 53209 renew

### PR DESCRIPTION
**Pull request by arykalin**

### CODE REVIEW

File: lib/objects/objects.rb [hunk 1]
```diff
@@ -7,8 +7,8 @@
 module Vcert
   SUPPORTED_CURVES = ["secp224r1", "prime256v1", "secp521r1"]
   class Request
-    attr_accessor :id
-    attr_reader :common_name, :country, :province, :locality, :organization, :organizational_unit, :san_dns,:key_type, :thumbprint
+    attr_accessor :id, :thumbprint
```
Comment (mr-tron) [inline lib/objects/objects.rb:10]: I think you can make it just optional argument in initialize function

File: lib/objects/objects.rb [hunk 2]
```diff
@@ -54,12 +58,16 @@ def generate_csr
...
       end
 
+      LOG.info("Making request from subject array #{subject_attrs.inspect}")
       subject = OpenSSL::X509::Name.new subject_attrs
       csr = OpenSSL::X509::Request.new
       csr.version = 0
       csr.subject = subject
       csr.public_key = @public_key
       if @san_dns != nil
+        if !@san_dns.kind_of?(Array)
```
Comment (mr-tron) [inline lib/objects/objects.rb]: `unless` instead `if !`

File: lib/tpp/tpp.rb [hunk 3]
```diff
@@ -265,7 +313,19 @@ def normalize_url(url)
...
+    Vcert::Certificate.new cert: pems[0], chain: pems[1..-1], private_key: nil
+  end
+
+  def search_by_thumbprint(thumbprint)
+    # thumbprint = re.sub(r'[^\dabcdefABCDEF]', "", thumbprint)
+    thumbprint = thumbprint.upcase
+    status, data = get(URL_CERTIFICATE_SEARCH+"?Thumbprint=#{thumbprint}")
+    # TODO: check that data have valid certificate in it
+    if status != 200
+      raise "Unexpected status code on Venafi Cloud certificate search. Status: #{status}. Message:\n #{data.body.to_s}"
```
Comment (mr-tron) [inline lib/tpp/tpp.rb]: `raise Vcert::ServerUnexpectedBehaviorError, "status #{status}"`

File: lib/vcert.rb [hunk 4]
```diff
@@ -56,26 +58,31 @@ def renew(*args)
...
+
+    def retrieve_loop(req, timeout: TIMEOUT)
       t = Time.new() + timeout
       loop do
         if Time.new() > t
           LOG.info("Waiting certificate #{req.id}")
           break
         end
-        certificate = retrieve(req)
+        certificate = @conn.retrieve(req)
```
Comment (mr-tron) [inline lib/vcert.rb:76]: don\`t need to specify `@conn.`

File: tests/test_vcert.rb [hunk 5]
```diff
@@ -10,7 +10,7 @@
 TPPUSER = ENV['TPPUSER']
 TPPPASSWORD = ENV['TPPPASSWORD']
 TPPZONE = ENV["TPPZONE"]
-TRUST_BUNDLE = "/tmp/chain.pem"
+TRUST_BUNDLE = ENV["TRUST_BUNDLE"]
```
Comment (mr-tron) [inline tests/test_vcert.rb:13]: In over products we use hardcoded value for tests

File: lib/vcert.rb (hunk 4 above)
Comment (arykalin) [reply lib/vcert.rb:76]: if you don't you will get  stack level too deep error, already tried

File: tests/test_vcert.rb (hunk 5 above)
Comment (arykalin) [reply tests/test_vcert.rb:13]: Everywhere in Jenkinsfile TRUST_BUNDLE is set from variables

File: lib/objects/objects.rb (hunk 1 above)
Comment (arykalin) [reply lib/objects/objects.rb:10]: :thumbprint is same as ID, you may need to write or read it after you created the request

File: lib/objects/objects.rb (hunk 2 above)
Comment (arykalin) [reply lib/objects/objects.rb]: done

File: lib/tpp/tpp.rb (hunk 3 above)
Comment (arykalin) [reply lib/tpp/tpp.rb]: done

**Pull request merged**